src/utils/benchmark.py

```diff
@@ -31,19 +34,35 @@ def benchmark_time(
     args = args if args is not None else ()
     kwargs = kwargs if kwargs is not None else {}
 
+    profiles_by_cls = {_cls: [] for _cls in profile_cls}
+
     for i in range(n):
         if setup is not None:
             args, kwargs = setup(*args, **kwargs)
 
-        with TimeContext() as t:
-            func(*args, **kwargs)
+        ctx_manager = ExitStack()
+
+        profiles = [ctx_manager.enter_context(Profiler(cls)) for cls in profile_cls]
+        with ctx_manager:
+            with TimeContext() as t:
+                func(*args, **kwargs)
+
+        for p in profiles:
+            profiles_by_cls[p._cls].append(p.profile)
 
         if teardown is not None:
             teardown()
 
         results.append(t.duration)
 
+    mean_profiles = []
+    for profile_cls, profile_list in profiles_by_cls.items():
+        mean_profiles.append(merge_profiles(profile_list))
+
     m = mean(results)
     s = stdev(results) if len(results) > 1 else None
```

Comment:
If there is no deviation I would put simply 0

Reply:
We are putting None already, I feel it is more correct, but we can change it in another PR